polaris-react/src/components/ActionMenu/components/Actions/Actions.tsx

```diff
@@ -154,23 +152,29 @@ export function Actions({actions = [], groups = [], onActionRollup}: Props) {
     actionsAndGroupsLengthRef.current = actionsAndGroups.length;
   }, [actions, groups, lastMenuGroup, lastMenuGroupWidth, onActionRollup]);
 
-  const handleResize = useMemo(
-    () =>
-      debounce(
-        () => {
-          if (!actionsLayoutRef.current) return;
-          availableWidthRef.current = actionsLayoutRef.current.offsetWidth;
-          // Set timesMeasured to 0 to allow re-measuring
-          timesMeasured.current = 0;
-          measureActions();
-        },
-        50,
-        {leading: false, trailing: true},
-      ),
-    [measureActions],
+  // don't use debounce with an inline function, we only want to create one debouncedMeasureActions
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedMeasureActions = useCallback(
+    debounce(
+      () => {
+        if (!actionsLayoutRef.current) return;
+        availableWidthRef.current = actionsLayoutRef.current.offsetWidth;
+        timesMeasured.current = 0;
+        measureActions();
+      },
+      50,
+      {leading: false, trailing: true},
+    ),
+    [measureActions, actionsLayoutRef],
   );
 
-  useEventListener('resize', handleResize);
+  const handleResize = useCallback(debouncedMeasureActions, [
+    debouncedMeasureActions,
+  ]);
+
+  useEventListener('resize', () => {
+    handleResize;
+  });
```

Comment:
The `handleResize` function isn't actually being called here. Might be a good idea to also pass through the event.

```suggestion
  useEventListener('resize', (event) => {
    handleResize(event);
  });
```